### ISSUE TITLE
Keep "Metrics to plot" element open after (de-)selecting a metric

### ISSUE THREAD

**Issue by Stan-vw** [piwik/piwik#21202]
## Summary

Personally I find it really frustrating that I need to click on the "Metrics to plot" icon for each and every goal I want to turn on/off. This is a doubling of clicks, as well as an extreme multiplication of required mouse movement.

My personal suggestion would be to keep it open until the user clicks somewhere outside of the element, but I'm interested to hear if anyone has a better suggestion (or why we should keep it the way it is).

Ideally the graph would update while turning on/off goals, without the need to close the element, but even if that's a requirement it feels like an improvement to me. In fairness, an argument against this would be that if you want to only make a single change, this would double the clicks required and add a bit of mouse movement (but only a little since it's only one-off)

Here's a short video that gives an idea of the extra actions needed to make a few changes: https://youtu.be/5YvMZyGRa50

**Comment by sgiehl** [comment]
I fully agree on that. Personally I would love to see our charting libraries being replaced with something more modern. That would also allows us to do some stuff like only updating the charts, without the need to full re-render the widget.
We could though only update the whole widget, once the users clicks somewhere else, but with the current implementation it's to much effort to let something already being loaded in the background.
From a UX point of view, I'm not sure if we also should add a "save"/"update" button to the selector, if it doesn't automatically update, as for some people it might be unexpected that they need to click somewhere outside...

**Comment by michalkleiner** [comment]
Relying on the blur event (click somewhere else to move the focus) is definitely not a standard way of triggering things, so an explicit "save/update" button would be great from that perspective.

**Comment by Stan-vw** [comment]
I can see the logic in that. Question though: if we add a Save/Update button, what happens if you just click outside of the element? If it saves, I'd find that really handy, but I'd also say it's weird that it saves/updates even though I didn't click on the save/update button

**Comment by michalkleiner** [comment]
Yeah, if we added a button it would ideally mean it wouldn't do any magic auto-update when clicking away, so it might be an extra step there.
Sometimes requiring an explicit save/update is good, but with a dropdown multiselect like this that opens on hover (no click needed) perhaps the blur event is enough. Other alternative might be to add a delay of say 1 or 2 seconds when it stays open after a selection to allow for clicking on other options, and if no click is registered within the limit it then closes automatically.

Also maybe thinking about the UX of this component from the keyboard user perspective might help us also improve it for mouse users?

Another option might be to add some mechanism that would allow the user do "select one and close" or "select multiple and confirm".

Too many options here, I think.
